scripts/generate/templates/SlotDerivation.js

```diff
@@ -0,0 +1,101 @@
+const format = require('../format-lines');
+const { TYPES } = require('./Slot.opts');
+
+const header = `\
+pragma solidity ^0.8.20;
+
+/**
+ * @dev Library for computing storage (and transient storage) locations from namespaces and deriving slots
+ * corresponding to standard patterns. The derivation method for array and mapping matches the storage layout used by
+ * the solidity language / compiler.
+ *
+ * See https://docs.soliditylang.org/en/v0.8.20/internals/layout_in_storage.html#mappings-and-dynamic-arrays[Solidity docs for mappings and dynamic arrays.].
+ */
+`;
+
+const namespace = `\
+/**
+ * @dev Derive an ERC-1967 slot from a string (namespace).
+ */
+function erc1967slot(string memory namespace) internal pure returns (bytes32 slot) {
```

Comment:
We write function names in camel case, shouldn't "slot" be capitalized?

```suggestion
function erc1967Slot(string memory namespace) internal pure returns (bytes32 slot) {
```